src/openrct2-ui/windows/Guest.cpp

```diff
@@ -1045,18 +1046,18 @@ void WindowGuestOverviewUpdate(rct_window* w)
 
     w->list_information_type += 2;
 
-    if ((w->highlighted_item & 0xFFFF) == 0xFFFF)
-        w->highlighted_item &= 0xFFFF0000;
-    else
-        w->highlighted_item++;
-
     // Disable peep watching thought for multiplayer as it's client specific
     if (network_get_mode() == NETWORK_MODE_NONE)
     {
-        // Create the "I have the strangest feeling I am being watched thought"
-        if ((w->highlighted_item & 0xFFFF) >= 3840)
+        if (w->guestWatchTimer == 0xFFFF)
```

Comment:
Unlike signed integers, unsigned integer overflow is well defined - if this timer was a `uint16_t` the code could safely let it wrap around to 0 without the need for this check.